### PR TITLE
SNOW-723810: Add coverage for SnowflakeDb classes

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-lf**
### Description
Regarding issue 113

The PR adds tests for SnowflakeDb classes to increase code coverage

### Checklist
- [x] Code compiles correctly
- [x] Code is formatted according to [Coding Conventions](../CodingConventions.md)
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing (`dotnet test`)
- [x] Extended the README / documentation, if necessary
- [x] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

File: Snowflake.Data.Tests/IntegrationTests/SFDbCommandIT.cs [hunk 1]
```diff
@@ -228,6 +246,22 @@ public void TestSimpleLargeResultSet()
...
+            {
+                // Arrange
+                conn.ConnectionString = ConnectionString;
+                conn.Open();
+
+                // Act
+                IDbCommand cmd = conn.CreateCommand();
+
+                // Assert
+                Assert.Throws<NotImplementedException>(() => cmd.Prepare());
```
Comment (sfc-gh-dstempniak) [inline Snowflake.Data.Tests/IntegrationTests/SFDbCommandIT.cs:262]: We shouldn't do integration test to test that Prepare throws an exception. It is very good candidate for unit tests.

Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/IntegrationTests/SFDbCommandIT.cs]: Sure, I'll move it to the unit tests

File: Snowflake.Data.Tests/IntegrationTests/SFDbCommandIT.cs [hunk 2]
```diff
@@ -203,6 +203,24 @@ public void TestSimpleCommand()
...
+        public void TestSimpleCommandWithConnectionAndCommandText()
+        {
+            using (IDbConnection conn = new SnowflakeDbConnection())
+            {
+                // Arrange
+                conn.ConnectionString = ConnectionString;
+                conn.Open();
+
+                // Act
+                IDbCommand cmd = new SnowflakeDbCommand((SnowflakeDbConnection)conn, "select 1");
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/IntegrationTests/SFDbCommandIT.cs]: I'll move this to the unit tests as well

File: Snowflake.Data.Tests/IntegrationTests/SFDbDataReaderIT.cs [hunk 3]
```diff
@@ -705,6 +746,38 @@ public void testGetBinary()
...
+                cmd.CommandText = $"insert into {TableName} values ('{testChar}')";
+                cmd.ExecuteNonQuery();
+                cmd.CommandText = $"select * from {TableName}";
+
+                // Act
+                using (IDataReader reader = cmd.ExecuteReader())
+                {
+                    // Assert
+                    Assert.IsTrue(reader.Read());
+                    Assert.IsTrue(testChar.Equals(reader.GetChar(0)));
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/IntegrationTests/SFDbDataReaderIT.cs]: Done

File: Snowflake.Data.Tests/IntegrationTests/SFDbTransactionIT.cs [hunk 4]
```diff
@@ -15,6 +15,42 @@ namespace Snowflake.Data.Tests.IntegrationTests
...
+                conn.Open();
+
+                // Act
+                using (IDbTransaction t1 = conn.BeginTransaction())
+                {
+                    // Assert
+                    Assert.AreEqual(IsolationLevel.ReadCommitted, t1.IsolationLevel);
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/IntegrationTests/SFDbTransactionIT.cs:52]: I tried without a real connection and it failed. The transaction needs an open connection when initializing

**Review by sfc-gh-dstempniak — APPROVED** [review]
LGTM

**Pull request merged**